scripts/cli/src/util/checkBridgeBalance.ts

```diff
@@ -0,0 +1,169 @@
+		user.forEach((data) => (userPolyxSum += data.polyx));
+		userPolyxSum + newPolyx;
+
+		checkPolyxLimit(newPolyx, offenderAddress, blockNumber);
+
+		if (userPolyxSum > BRIDGE_POLY_TOTAL) {
+			console.log(`Error: Total POLYX Bridge limit exceeded.`);
+			console.log(
+				"Polyx Total Bridge Limit Equivocation: ",
+				offenderAddress,
+				blockNumber
+			);
+			prom.polyxTotalBridgeLimitEquivocations.inc({ offenderAddress });
+		} else {
+			console.log("Updates user data.");
+			user.push({
+				polyx: newPolyx,
+				lastTimeBridged: currentTimeToSeconds(),
+			});
+			userMap.set(offenderAddress, user);
+		}
+	}
+
+	function removeOldTx(
+		offenderAddress: string,
+		user: userInfo[],
+		userMap: Map<string, userInfo[]>
+	) {
+		const validTxArray = user.filter((data) => {
+			const timeDifference = currentTimeToSeconds() - data.lastTimeBridged;
+			return timeDifference < BRIDGE_TIME_PERIOD;
+		});
+		userMap.set(offenderAddress, validTxArray);
+	}
+
+	function checkPolyxLimit(
+		newPolyx: number,
+		offenderAddress: string,
+		blockNumber: number
+	) {
+		if (newPolyx > BRIDGE_LIMIT) {
+			console.log(`Error: Transaction POLYX limit exceeded.`);
+			console.log(
+				"Polyx Bridge Limit Equivocation: ",
+				offenderAddress,
+				blockNumber
+			);
+			prom.polyxBridgeLimitEquivocations.inc({ offenderAddress });
+		}
+	}
+
+	// Subscribe to the new headers on-chain. The callback is fired when new headers
+	// are found, the call itself returns a promise with a subscription that can be
+	// used to unsubscribe from the newHead subscription
+	const unsubscribe = await api.rpc.chain.subscribeNewHeads((header) => {
+		console.log(`Chain is at block: #${header.number}`);
+
+		// Subscribe to system events via storage
+		api.query.system.events((events) => {
+			console.log(`\nReceived ${events.length} events:`);
+
+			events
+				.map((record) => {
+					return record.event;
+				})
+				.filter((event) => {
+					return (
+						event.section === "bridge" && event.method === "BridgeTxScheduled"
+					);
+				})
+				.forEach((event) => {
+					const types = event.typeDef;
+
+					event.data
+						.filter((data, index) => types[index].type === "BridgeTx")
+						.forEach((data) => {
+							const txData = JSON.parse(data.toString());
+							const newPolyx = txData.value;
+							const offenderAddress = txData.recipient;
+							const userExist = userTxs.get(offenderAddress);
+
+							if (userExist) {
+								console.log("User exists.");
+								checksUser(
+									userTxs,
+									newPolyx,
+									offenderAddress,
+									header.number.toNumber()
+								);
+							} else {
+								console.log("User hasn't been added to array yet.");
+								checkPolyxLimit(
+									newPolyx,
+									offenderAddress,
+									header.number.toNumber()
+								);
+
+								userTxs.set(offenderAddress, [
+									{
+										polyx: newPolyx,
+										lastTimeBridged: currentTimeToSeconds(),
+									},
+								]);
+							}
```

Comment:
If the user doesn't exist, just initialize their entry to empty array.  Then you don't need the if/else here and can just call `checkUser`.

```
if (!userTxs.has(offenderAddress)) {
  // New user.
  userMap.set(offenderAddress, []);
}
checksUser(userTxs, newPolyx, offenderAddress, header.number.toNumber());
```